code/code/misc/other.cc

```diff
@@ -4175,20 +4173,27 @@ void Descriptor::send_feedback(const char *subject, const char *msg)
   message += "\n";
 
   // could use vsystem, but we want the file i/o outside this thread as well
-  if (0 != vfork())
-  {
-    FILE *fp;
-    sstring tempfile = format("/usr/tmp/feedback%d.tmp") % tempInc++;
-    if ((fp = fopen(tempfile.c_str(), "w")))
-    {
-        fputs(message.c_str(), fp);
-        fclose(fp);
-        int err = system((format("/usr/sbin/sendmail -t < %s") % tempfile).str().c_str());
-        if (err)
-            vlogf(LOG_MISC, format("Call to sendmail returned nonzero status %d") % err);
-        remove(tempfile.c_str());
+  if (!vfork()) {
+    char tempfile[32];
+    strcpy(tempfile, "/tmp/sneezy_sendmail_XXXXXX.tmp");
```

Comment:
Why not `char tempfile[] = "/tmp/";`

Reply:
mkstemps requires a full template to work with. it replaces the Xes with random values and then securely creates a new temp file from the resulting path.

Reply:
Why can't we use a string literal to give it an initial value?

Reply:
mkstemps would then be trying to modify the RO string literal, no? I'm sure there's some handy C++ way to initialize the array with a copy of the literal instead of just pointing to it, but everyone seems to be using strcpy...

Reply:
You're declaring a `char[]`, not a `const char*`, so it'll get its own copy. This works fine:

```cpp
int main() {
  char str[] = "bla\n";
  str[0] = 'f';
  cout << str;
}
```

Reply:
Ahhh forgot about that detail. Well, gonna redo this PR anyway so it don't matter. Thanks for being so patient.